clang/lib/Index/IndexRecordHasher.cpp

```diff
@@ -301,6 +291,18 @@ static hash_code computeHash(const IdentifierInfo *II) {
   return hash_value(II->getName());
 }
 
+hash_code IndexRecordHasher::hashImpl(MacroDef MD) {
+  hash_code Hash = INITIAL_HASH;
+  COMBINE_HASH(StringRef("@macro@"), computeHash(MD.Name));
+  auto &SM = getASTContext().getSourceManager();
+  auto Loc = MD.MI->getDefinitionLoc();
+  // Only hash the location if it's not in a system header, to match how
+  // USR generation behaves.
+  if (SM.isInSystemHeader(Loc))
+    COMBINE_HASH(hashLoc(SM, Loc, /*IncludeOffset=*/true));
```

Comment:
Huh, I thought `record-hash-macros.c` would cover the location hash since it defines the same macro in different places. I'll have to dig into why it was still creating two records.  Thanks!

Reply:
Oh I get it: we hash the location of the occurrence regardless of the hashing of the macro itself. To make the test catch this I need to move the definition into a header file.